File: BE/CloudTicTacToeBackend/CloudTicTacToe.Domain/Models/GameBoard.cs
using CloudTicTacToe.Domain.Enums;

namespace CloudTicTacToe.Domain.Models
{
    public class GameBoard : BaseDomainModel
    {
        public const int BOARD_SIZE = 3;
        public Player PlayerX { get; set; } = null!;
        public Player? PlayerO { get; set; }
        public GameGoardState State { get; set; }
        public Guid NextPlayerId { get; set; }
        public IEnumerable<Cell> Cells { get; set; } = null!;
    }
}

File: FE/CloudTicTacToe/src/app/models/game-board.ts
import { Cell } from "./cell";
import { Player } from "./player";

export interface GameBoard {
  id: string;
  state: 'Waiting' | 'Ongoing' | 'Draw' | 'WinnX' | 'WinnO';
  board: Cell[][];
  nextPlayerId: string;
  playerX: Player | null;
  playerO: Player | null;
}
